### ISSUE TITLE
Error when trying to triangulate using 3D and multianimal

### ISSUE THREAD

**Issue by karipan** [DeepLabCut/DeepLabCut#1312]
**My Operating system and DeepLabCut version**
Ubuntu18.04.5 LTS, Google Colab & DeepLabCut2.1.10.4

**Describe the problem**
I'm using 3D mode(2 cameras) and multi animal mode. 
I trained and evaluated network about each camera model.
Then I analyzed and filtered them.
And now, I want to triangulate and integrate them, but an error occurred.

<details><summary>Code output</summary><p>

```
Config:
{'all_joints': [[0], [1], [2], [3]],
 'all_joints_names': ['point1', 'root1', 'point2', 'root2'],
 'batch_size': 1,
 'crop_pad': 0,
 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_PreferenceTestJun6/PreferenceTest_nelableft95shuffle1.pickle',
 'dataset_type': 'multi-animal-imgaug',
 'deterministic': False,
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'init_weights': '/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 1.0,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'mean_pixel': [123.68, 116.779, 103.939],
 'minconfidence': 0.01,
 'mirror': False,
 'net_type': 'resnet_50',
 'nmsradius': 5.0,
 'num_joints': 4,
 'num_limbs': 3,
 'optimizer': 'sgd',
 'pairwise_huber_loss': True,
 'pairwise_predict': False,
 'partaffinityfield_graph': [[0, 1], [2, 3], [1, 3]],
 'partaffinityfield_predict': True,
 'regularize': False,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': '/content/drive/MyDrive/4S/PreferenceTest-nelableft-2021-06-06/dlc-models/iteration-0/PreferenceTestJun6-trainset95shuffle1/test/snapshot',
 'stride': 8.0,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
List of pairs: [['/content/drive/MyDrive/4S/videos/left_hand_0530_2.mp4', '/content/drive/MyDrive/4S/videos/right_hand_0530_2.mp4'], ['/content/drive/MyDrive/4S/videos/left_hand_0530_3.mp4', '/content/drive/MyDrive/4S/videos/right_hand_0530_3.mp4'], ['/content/drive/MyDrive/4S/videos/left_hand_0530_1.mp4', '/content/drive/MyDrive/4S/videos/right_hand_0530_1.mp4']]
Analyzing video /content/drive/MyDrive/4S/videos/left_hand_0530_2.mp4 using config_file_left
Using snapshot-10000 for model /content/drive/MyDrive/4S/PreferenceTest-nelableft-2021-06-06/dlc-models/iteration-0/PreferenceTestJun6-trainset95shuffle1
Initializing ResNet
Activating extracting of PAFs
Starting to analyze %  /content/drive/MyDrive/4S/videos/left_hand_0530_2.mp4
/content/drive/MyDrive/4S/videos  already exists!
Video already analyzed! /content/drive/MyDrive/4S/videos/left_hand_0530_2DLC_resnet50_PreferenceTestJun6shuffle1_10000.h5
The videos are analyzed. Time to assemble animals and track 'em... 
 Call 'create_video_with_all_detections' to check multi-animal detection quality before tracking.
If the tracking is not satisfactory for some videos, consider expanding the training set. You can use the function 'extract_outlier_frames' to extract a few representative outlier frames.
/content/drive/MyDrive/4S/videos left_hand_0530_2 DLC_resnet50_PreferenceTestJun6shuffle1_10000
Filtering with median model /content/drive/MyDrive/4S/videos/left_hand_0530_2.mp4
Data from left_hand_0530_2 were already filtered. Skipping...
Analyzing video /content/drive/MyDrive/4S/videos/right_hand_0530_2.mp4 using config_file_right
Using snapshot-8000 for model /content/drive/MyDrive/4S/PreferenceTest-nelabright-2021-06-06/dlc-models/iteration-0/PreferenceTestJun6-trainset95shuffle1
Initializing ResNet
Activating extracting of PAFs
Starting to analyze %  /content/drive/MyDrive/4S/videos/right_hand_0530_2.mp4
/content/drive/MyDrive/4S/videos  already exists!
Video already analyzed! /content/drive/MyDrive/4S/videos/right_hand_0530_2DLC_resnet50_PreferenceTestJun6shuffle1_8000.h5
The videos are analyzed. Time to assemble animals and track 'em... 
 Call 'create_video_with_all_detections' to check multi-animal detection quality before tracking.
If the tracking is not satisfactory for some videos, consider expanding the training set. You can use the function 'extract_outlier_frames' to extract a few representative outlier frames.
/content/drive/MyDrive/4S/videos right_hand_0530_2 DLC_resnet50_PreferenceTestJun6shuffle1_8000
Filtering with median model /content/drive/MyDrive/4S/videos/right_hand_0530_2.mp4
Data from right_hand_0530_2 were already filtered. Skipping...
Undistorting...
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-44-53fdbd661d57> in <module>()
      1 config_path3d = "/content/drive/MyDrive/4S/PreferenceTest-nelab-2021-06-06-3d/config.yaml"
----> 2 deeplabcut.triangulate(config_path3d, '/content/drive/MyDrive/4S/videos', videotype='.mp4', filterpredictions=True, save_as_csv=True)
      3 # filterpredictions=True/False

2 frames
/usr/local/lib/python3.7/dist-packages/pandas/io/pytables.py in read_hdf(path_or_buf, key, mode, errors, where, start, stop, columns, iterator, chunksize, **kwargs)
    385 
    386         if not exists:
--> 387             raise FileNotFoundError(f"File {path_or_buf} does not exist")
    388 
    389         store = HDFStore(path_or_buf, mode=mode, errors=errors, **kwargs)

FileNotFoundError: File /content/drive/MyDrive/4S/videos/left_hand_0530_2DLC_resnet50_PreferenceTestJun6shuffle1_10000.h5 does not exist
```

</p></details>

I renamed left_hand_0530_2DLC_resnet50_PreferenceTestJun6shuffle1_10000_el_filtered.h5 
or left_hand_0530_2DLC_resnet50_PreferenceTestJun6shuffle1_10000_el.h5
as left_hand_0530_2DLC_resnet50_PreferenceTestJun6shuffle1_10000.h5.
However, another error occurred.

<details><summary>Code output</summary><p>
```
Config:
{'all_joints': [[0], [1], [2], [3]],
 'all_joints_names': ['point1', 'root1', 'point2', 'root2'],
 'batch_size': 1,
 'crop_pad': 0,
 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_PreferenceTestJun6/PreferenceTest_nelableft95shuffle1.pickle',
 'dataset_type': 'multi-animal-imgaug',
 'deterministic': False,
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'init_weights': '/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 1.0,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'mean_pixel': [123.68, 116.779, 103.939],
 'minconfidence': 0.01,
 'mirror': False,
 'net_type': 'resnet_50',
 'nmsradius': 5.0,
 'num_joints': 4,
 'num_limbs': 3,
 'optimizer': 'sgd',
 'pairwise_huber_loss': True,
 'pairwise_predict': False,
 'partaffinityfield_graph': [[0, 1], [2, 3], [1, 3]],
 'partaffinityfield_predict': True,
 'regularize': False,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': '/content/drive/MyDrive/4S/PreferenceTest-nelableft-2021-06-06/dlc-models/iteration-0/PreferenceTestJun6-trainset95shuffle1/test/snapshot',
 'stride': 8.0,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
List of pairs: [['/content/drive/MyDrive/4S/videos/left_hand_0530_1.mp4', '/content/drive/MyDrive/4S/videos/right_hand_0530_1.mp4'], ['/content/drive/MyDrive/4S/videos/left_hand_0530_2.mp4', '/content/drive/MyDrive/4S/videos/right_hand_0530_2.mp4'], ['/content/drive/MyDrive/4S/videos/left_hand_0530_3.mp4', '/content/drive/MyDrive/4S/videos/right_hand_0530_3.mp4']]
Analyzing video /content/drive/MyDrive/4S/videos/left_hand_0530_1.mp4 using config_file_left
Using snapshot-10000 for model /content/drive/MyDrive/4S/PreferenceTest-nelableft-2021-06-06/dlc-models/iteration-0/PreferenceTestJun6-trainset95shuffle1
Initializing ResNet
Activating extracting of PAFs
Config:
{'all_joints': [[0], [1], [2], [3]],
 'all_joints_names': ['point1', 'root1', 'point2', 'root2'],
 'batch_size': 1,
 'crop_pad': 0,
 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_PreferenceTestJun6/PreferenceTest_nelabright95shuffle1.pickle',
 'dataset_type': 'multi-animal-imgaug',
 'deterministic': False,
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'init_weights': '/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 1.0,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'mean_pixel': [123.68, 116.779, 103.939],
 'minconfidence': 0.01,
 'mirror': False,
 'net_type': 'resnet_50',
 'nmsradius': 5.0,
 'num_joints': 4,
 'num_limbs': 3,
 'optimizer': 'sgd',
 'pairwise_huber_loss': True,
 'pairwise_predict': False,
 'partaffinityfield_graph': [[0, 1], [2, 3], [1, 3]],
 'partaffinityfield_predict': True,
 'regularize': False,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': '/content/drive/MyDrive/4S/PreferenceTest-nelabright-2021-06-06/dlc-models/iteration-0/PreferenceTestJun6-trainset95shuffle1/test/snapshot',
 'stride': 8.0,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
Starting to analyze %  /content/drive/MyDrive/4S/videos/left_hand_0530_1.mp4
/content/drive/MyDrive/4S/videos  already exists!
Video already analyzed! /content/drive/MyDrive/4S/videos/left_hand_0530_1DLC_resnet50_PreferenceTestJun6shuffle1_10000.h5
The videos are analyzed. Time to assemble animals and track 'em... 
 Call 'create_video_with_all_detections' to check multi-animal detection quality before tracking.
If the tracking is not satisfactory for some videos, consider expanding the training set. You can use the function 'extract_outlier_frames' to extract a few representative outlier frames.
/content/drive/MyDrive/4S/videos left_hand_0530_1 DLC_resnet50_PreferenceTestJun6shuffle1_10000
Filtering with median model /content/drive/MyDrive/4S/videos/left_hand_0530_1.mp4
2 possible data files were found: ['left_hand_0530_1DLC_resnet50_PreferenceTestJun6shuffle1_10000.h5', 'left_hand_0530_1DLC_resnet50_PreferenceTestJun6shuffle1_10000_el.h5'].
Picking the first by default...
Saving filtered csv poses!
Analyzing video /content/drive/MyDrive/4S/videos/right_hand_0530_1.mp4 using config_file_right
Using snapshot-8000 for model /content/drive/MyDrive/4S/PreferenceTest-nelabright-2021-06-06/dlc-models/iteration-0/PreferenceTestJun6-trainset95shuffle1
Initializing ResNet
Activating extracting of PAFs
Starting to analyze %  /content/drive/MyDrive/4S/videos/right_hand_0530_1.mp4
/content/drive/MyDrive/4S/videos  already exists!
Video already analyzed! /content/drive/MyDrive/4S/videos/right_hand_0530_1DLC_resnet50_PreferenceTestJun6shuffle1_8000.h5
The videos are analyzed. Time to assemble animals and track 'em... 
 Call 'create_video_with_all_detections' to check multi-animal detection quality before tracking.
If the tracking is not satisfactory for some videos, consider expanding the training set. You can use the function 'extract_outlier_frames' to extract a few representative outlier frames.
/content/drive/MyDrive/4S/videos right_hand_0530_1 DLC_resnet50_PreferenceTestJun6shuffle1_8000
Filtering with median model /content/drive/MyDrive/4S/videos/right_hand_0530_1.mp4
Data from right_hand_0530_1 were already filtered. Skipping...
Undistorting...
0it [00:00, ?it/s]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-37-53fdbd661d57> in <module>()
      1 config_path3d = "/content/drive/MyDrive/4S/PreferenceTest-nelab-2021-06-06-3d/config.yaml"
----> 2 deeplabcut.triangulate(config_path3d, '/content/drive/MyDrive/4S/videos', videotype='.mp4', filterpredictions=True, save_as_csv=True)
      3 # filterpredictions=True/False

3 frames
/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_3d/triangulation.py in triangulate(config, video_path, videotype, filterpredictions, filtertype, gputouse, destfolder, save_as_csv)
    303                 path_stereo_file,
    304             ) = undistort_points(
--> 305                 config, dataname, str(cam_names[0] + "-" + cam_names[1])
    306             )
    307             if len(dataFrame_camera1_undistort) != len(dataFrame_camera2_undistort):

/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_3d/triangulation.py in undistort_points(config, dataframe, camera_pair)
    502                 scorer_cam1, bp, "likelihood"
    503             ] = dataframe_cam1.xs(
--> 504                 [bp, "likelihood"], level=["bodyparts", "coords"], axis=1
    505             ).values
    506 

/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py in __setitem__(self, key, value)
    668 
    669         iloc = self if self.name == "iloc" else self.obj.iloc
--> 670         iloc._setitem_with_indexer(indexer, value)
    671 
    672     def _validate_key(self, key, axis: int):

/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py in _setitem_with_indexer(self, indexer, value)
   1726                     if len(ilocs) != value.shape[1]:
   1727                         raise ValueError(
-> 1728                             "Must have equal len keys and value "
   1729                             "when setting with an ndarray"
   1730                         )

ValueError: Must have equal len keys and value when setting with an ndarray
```

</p></details>

**How to Reproduce the problem**
I think it's impossible to reproduce all flows because Google Colab cannot use GUI. So I omitted most part.  I'm sorry.

<details><summary>My Code about "right" camera</summary><p>

```
config_path = '/content/drive/MyDrive/4S/PreferenceTest-nelabright-2021-06-06/config.yaml'

path1='/content/drive/MyDrive/4S/videos/right_hand_0530_1.mp4'
path2='/content/drive/MyDrive/4S/videos/right_hand_0530_2.mp4'
path3='/content/drive/MyDrive/4S/videos/right_hand_0530_3.mp4'
scorername = deeplabcut.analyze_videos(config_path,[path1, path2, path3], videotype='.mp4')
deeplabcut.create_video_with_all_detections(config_path, [path1, path2, path3], scorername)

deeplabcut.analyze_videos(config_path,[path1, path2, path3], videotype='.mp4', save_as_csv=True)

deeplabcut.convert_detections2tracklets(config_path, [path1, path2, path3], videotype='mp4',
                                        shuffle=1, trainingsetindex=0, track_method='ellipse')

deeplabcut.filterpredictions(config_path, [path1, path2, path3], videotype='mp4', save_as_csv=True, track_method="ellipse")

pickle1 = '/content/drive/MyDrive/4S/videos/right_hand_0530_1DLC_resnet50_PreferenceTestJun6shuffle1_8000_el.pickle'
pickle2 = '/content/drive/MyDrive/4S/videos/right_hand_0530_2DLC_resnet50_PreferenceTestJun6shuffle1_8000_el.pickle'
pickle3 = '/content/drive/MyDrive/4S/videos/right_hand_0530_3DLC_resnet50_PreferenceTestJun6shuffle1_8000_el.pickle'
deeplabcut.refine_training_dataset.convert_raw_tracks_to_h5(config_path, pickle1)
deeplabcut.refine_training_dataset.convert_raw_tracks_to_h5(config_path, pickle2)
deeplabcut.refine_training_dataset.convert_raw_tracks_to_h5(config_path, pickle3)

deeplabcut.plot_trajectories(config_path,[path1, path2, path3],filtered = True, track_method='ellipse')
```

</p></details>

The same proccess was done about "left" camera.
Then, I tried to run this:

config_path3d = "/content/drive/MyDrive/4S/PreferenceTest-nelab-2021-06-06-3d/config.yaml"
deeplabcut.triangulate(config_path3d, '/content/drive/MyDrive/4S/videos', videotype='.mp4', filterpredictions=True, save_as_csv=True)

If more information were needed, please ask me. I'm not accustomed to this type of question.
And, sorry for my poor English skills!

**Comment by jeylau** [comment]
Hi @karin-oshima, could you tell me what pandas version you're using?

**Comment by karipan** [comment]
I recently renewed my env and I don't remember its version(sorry), but probably it was 1.1.3, I think.

In addition, I found the reason why "ValueError: Must have equal len keys and value when setting with an ndarray" occurred to me.

In  undistort_points(config, dataframe, camera_pair) from "/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_3d/triangulation.py", 
the shape of dataFrame_cam1_undistort is not adapted to multi-animal mode.

Thank you!
